### PR TITLE
Add token authentication feature and adjust tests

### PR DESCRIPTION
**Pull request by caiolombello**
This pull request addresses issue #4: Accesstoken factory and connection id by introducing a new token-based authentication mechanism to enhance the security and flexibility of our application. The key changes include:

- **Token Generation:** Implemented a secure method for generating authentication tokens upon successful user login.
- **Token Validation:** Added middleware to validate tokens for protected routes, ensuring only authenticated users can access these resources.
- **User Roles and Permissions:** Incorporated user roles and permissions to further control access to different parts of the application.
- **Token Expiry:** Configured token expiry to ensure tokens are valid only for a specified period, enhancing security.
- **Testing Adjustments:** Updated existing tests and added new ones to cover the token authentication functionality.
- **Documentation:** Updated the README and API documentation to include details on how to use the new token-based authentication system.
- **Test Results:** All tests have been run and passed successfully.

These changes aim to provide a more robust and scalable authentication system, improving overall security and user experience. Please review the implementation and provide feedback or request changes as needed.

### CODE REVIEW

**Comment by caiolombello** [comment]
Hey there! I've just completed the Roadmap with the following tasks:

- [x] More documentation, both internal and user.
- [x] Integration tests with containerized ASP hello-world server.
- [x] Ensure that authentication works correctly.

I've thoroughly tested all these features, ensuring everything runs smoothly, including tests conducted using a containerized ASP server.

**Review by droserasprout — CHANGES_REQUESTED** [review]
This is one of the greatest PR I've ever seen! Thank you so much! I have just a couple of nitpicks, see the comments.

File: AspNetAuthExample/Hub/WeatherHub.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+using Microsoft.AspNetCore.SignalR;
+using System.Threading.Tasks;
+
+using AspNetAuthExample.Controllers;
+
+namespace AspNetAuthExample
+{
+    public class WeatherHub : Hub
+    {
+        // Método para enviar uma mensagem a todos os clientes conectados
```
[the remark this answers is not in the document]
Comment (caiolombello) [reply AspNetAuthExample/Hub/WeatherHub.cs]: Changed at [929399b](https://github.com/baking-bad/pysignalr/pull/15/commits/929399bfadbe67753174f97a5d1f186947b2866b)

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Code of Conduct
```
[the remark this answers is not in the document]
Comment (caiolombello) [reply CODE_OF_CONDUCT.md]: Removed at [9d4e3b3](https://github.com/baking-bad/pysignalr/pull/15/commits/9d4e3b39528fe3d2267bf43f2d5ef5463fa4186f)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+```
+
+2. Execute the test suite by running:
+```bash
+poetry run make test
+```
+
+4. After making changes, ensure all tests pass and that the coverage is maintained or improved.
+
+## Code of Conduct
```
[the remark this answers is not in the document]
Comment (caiolombello) [reply CONTRIBUTING.md]: Removed at [9d4e3b3](https://github.com/baking-bad/pysignalr/pull/15/commits/9d4e3b39528fe3d2267bf43f2d5ef5463fa4186f)

File: README.md [hunk 4]
```diff
@@ -60,8 +77,91 @@ with suppress(KeyboardInterrupt, asyncio.CancelledError):
...
+#### Attributes
+
+- `error` (str): The error message, if any.
+
 ## Roadmap
 
-- [ ] More documentation, both internal and user.
-- [ ] Integration tests with containerized ASP hello-world server.
-- [ ] Ensure that authentication works correctly.
+- [x] More documentation, both internal and user.
```
[the remark this answers is not in the document]
Comment (caiolombello) [reply README.md]: Removed Roadmap section at [6200346](https://github.com/baking-bad/pysignalr/pull/15/commits/620034635ca92ad0f1b0e82f4e2873cbbe996539)

File: pyproject.toml [hunk 5]
```diff
@@ -67,6 +67,11 @@ ignore = [
 target-version = "py38"
 
 [tool.ruff.lint]
+ignore = [
```
[the remark this answers is not in the document]
Comment (caiolombello) [reply pyproject.toml:71]: Added at [e172fc5](https://github.com/baking-bad/pysignalr/pull/15/commits/e172fc5886cf8bf7f04c7e7433bee7c80f2da7c5). Thanks @droserasprout

**Review by droserasprout — APPROVED** [review]

**Comment by droserasprout** [comment]
I'm going to fix CI and a couple of things, then release 1.1.0 with our changes.

**Comment by caiolombello** [comment]
I noticed that the pipeline failed due to the lack of Docker Compose. I've never worked with GitHub Pipeline before, so I need some help with it.

**Pull request merged**